src/Illuminate/Filesystem/AwsS3V3Adapter.php

```diff
@@ -95,6 +95,35 @@ public function temporaryUrl($path, $expiration, array $options = [])
         return (string) $uri;
     }
 
+    /**
+     * Get a temporary upload URL for the file at the given path.
+     *
+     * @param  string  $path
+     * @param  \DateTimeInterface  $expiration
+     * @param  array  $options
+     * @return string
+     */
+    public function temporaryUploadUrl($path, $expiration, array $options = [])
+    {
+        $command = $this->client->getCommand('PutObject', array_merge([
+            'Bucket' => $this->config['bucket'],
+            'Key' => $this->prefixer->prefixPath($path),
+        ], $options));
+
+        $uri = $this->client->createPresignedRequest(
+            $command, $expiration, $options
+        )->getUri();
+
+        // If an explicit base URL has been set on the disk configuration then we will use
+        // it as the base URL instead of the default path. This allows the developer to
+        // have full control over the base path for this filesystem's generated URLs.
+        if (isset($this->config['temporary_url'])) {
+            $uri = $this->replaceBaseUrl($uri, $this->config['temporary_url']);
+        }
```

Comment:
Is this block of code actually needed for temporary upload URLs? Why or why not?

Reply:
As we all know, `temporary_url` option of `disks` config in `config/filesystems.php` file is used to override the host/base url of the generated temporary URLs.

`temporaryUrl()` already included this logic of `replaceBaseUrl()`, So I included it too in my proposed `temporaryUploadUrl()` method.

As you can see, `temporaryUploadUrl()` generated a URL with my custom host i.e. '**example.com**' since I set it up in the **temporary_url** config.

![image](https://user-images.githubusercontent.com/30714223/213932154-06873ae9-3ba8-4474-9445-6e83ffe369ee.png)